### DISCUSSION TITLE
Progress-bar is not working smoothly

### DISCUSSION THREAD

**Discussion by sercandemirtas** [google/model-viewer#2371] [Q&A]
Hello all!

Im having problem with progress-bar. As you can see in demo link and first video progress-bar is not working properly. I mean first it is coming to %50-%60 then it goes to %100. Is it possible to make it smooth ?

Any suggestion or help would be great. Thanks...

#### Live Demo
www.basebros.com/ar-projects.html

This is my progress bar

https://user-images.githubusercontent.com/68244546/115105776-ac896500-9f69-11eb-972d-61e7654b0e9c.mov

This is what I want to do 

https://user-images.githubusercontent.com/68244546/115105784-b9a65400-9f69-11eb-835d-df76cce424a2.mov

**My codes which I used in live demo and first video.**

  const modelViewer = document.querySelector('model-viewer');
  const progress = document.querySelector('#progress');
  const bar = progress.querySelector('.bar');

  modelViewer.addEventListener('progress', (event) => {
    const {totalProgress} = event.detail;
    progress.classList.toggle('show', totalProgress < 1);
    bar.style.transform = `scaleX(${totalProgress})`;
  });

  #progress {
    position: absolute;
    width: calc(100% / 3);
    height: 10px;
    color: rgba(255, 255, 255, 0.952);
    box-shadow: 0px 0px 8px 6px rgba(24, 24, 24, 0.534);
    border: 2px solid currentColor;
    border-radius: 10px;
    top: calc(50% - 5px);
    left: calc(50% - 100% / 6);
    opacity: 0;
    transition: opacity 0.3s 0.3s;
  }

  #progress.show {
    transition-delay: 0s;
    opacity: 1;
  }

  #progress>.bar {
    width: 100%;
    height: 100%;
    background-color: currentColor;
    transform-origin: top left;
    transform: scaleX(0);
  }

### Version
- model-viewer: v1.6

### Browser Affected

- [ x] Chrome, version: xx.x.xxxx.xx
- [ x] Edge
- [ x] Firefox
- [ x] IE
- [ x] Safari

### OS

- [ x] Android
- [ x] iOS
- [ x] Linux
- [x ] MacOS
- [ x] Windows

**Comment by elalish** [comment]
v1.1 is quite old; can you try with the latest, v1.6? This may have already been fixed. Also, does the default progress bar have the same behavior?

**Comment by sercandemirtas** [comment]
@elalish I updated to v1.6 it is still same (you can try on demo link) and also I tried default progress bar, same behavior again. 
Is this problem just on me for both custom and default progress bar or anyone having this problem too ?

**Comment by elalish** [comment]
We've seen this occasionally. The progress is actually sent by the server as part of the file fetch process. Sometimes the server config is messed up so that it either doesn't send progress at all, or sends the wrong values.

**Comment by Aztec91** [comment]
Also a funny note. If you click on the bar while the model loads, it fills faster (especially, when you have large file-sizes)

**Comment by elalish** [comment]
I'm guessing that has less to do with clicking and more to do with browser caching.
